context/app/static/js/components/workspaces/ConfirmDeleteWorkspacesDialog/ConfirmDeleteWorkspacesDialog.tsx

```diff
@@ -0,0 +1,86 @@
+import React from 'react';
+
+import {
+  Box,
+  Button,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogTitle,
+  Divider,
+  IconButton,
+  Stack,
+} from '@mui/material';
+import CloseRounded from '@mui/icons-material/CloseRounded';
+
+import { useSnackbarActions } from 'js/shared-styles/snackbars';
+import { SelectedItems } from 'js/hooks/useSelectItems';
+import { generateCommaList } from 'js/helpers/functions';
+
+import { MergedWorkspace } from '../types';
+
+interface ConfirmDeleteWorkspacesDialogProps {
+  dialogIsOpen: boolean;
+  handleClose: () => void;
+  handleDeleteWorkspace: (workspaceId: number) => Promise<void>;
+  selectedWorkspaceIds: SelectedItems;
+  workspacesList: MergedWorkspace[];
+}
+export default function ConfirmDeleteWorkspacesDialog({
+  dialogIsOpen,
+  handleClose,
+  handleDeleteWorkspace,
+  selectedWorkspaceIds,
+  workspacesList,
+}: ConfirmDeleteWorkspacesDialogProps) {
+  const { toastError } = useSnackbarActions();
+
+  const handleDeleteAndClose = () => {
+    const workspaceIds = [...selectedWorkspaceIds];
+
+    Promise.all(workspaceIds.map((workspaceId) => handleDeleteWorkspace(Number(workspaceId)))).catch((e) => {
+      toastError(`Error deleting workspaces: ${generateCommaList(workspaceIds)}`);
+      console.error(e);
+    });
+
+    selectedWorkspaceIds.clear();
```

Comment:
Do we want to only clear the selected ids if the delete was successful? Should we also add a `toastSuccess`?

Reply:
That's a good point! Adding a success indicator is also a good idea.